### PR TITLE
Add server env tab

### LINKED ISSUE #146: Add fields for Server Environments : Sandbox, Development, QA, and Production.
Description : We need a system for tracking all of information around the different environments from Sandbox to Production for multi-tiered stacked application.  Sys Ops should be able to see all of the information around Server Environments in one TAB on the Software Record Show and Edit Page.  The fields are listed below.

Depends ON : #206 

Done looks like

- [x] Server Environment Tab
- [x] 8 fields listed below on the TAB
- [x] All show and edit pages.

### LINKED ISSUE #224: Move Yes-No toggle value calls out of controller
In `/app/controllers/software_records_controller.rb` we have a line to fetch the value of a record attribute for use in a yes/no radio button toggle for use in the view. 

[@software_sc = SoftwareRecord.find_by(id: params[:id]).support_contract](https://github.com/uclibs/application_portfolio/blob/4bd53a1fa294c1ffbabdc17b273e6fafbea539d8/app/controllers/software_records_controller.rb#L74)

As the number of these toggle variables grows, we'll need a more sustainable pattern. Look at ways to either eliminate this call or consolidate to a generalized function in a helper.

### PR DESCRIPTION
**Pull request by crowesn**
Resolves #146 
Resolves #224 

Add server environment form and show tabs.
Break out show tabs into partials.
Move yes|no toggle var to helper.

### CODE REVIEW

File: app/views/software_records/_form_server_environment.html.erb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            </div>
+
+            <div class="form-group">
+              <%= form.label "last_record_change" %>
+               <%= form.date_field :last_record_change, as: :date, value: form.object.try(:strftime,"%m/%d/%Y"), class: 'form-control' %>
+            </div>
+
+            <% if params[:id].nil? %>
+              <div class="form-group">
+                <%= form.label :track_uptime, 'Track uptime?' %><br/>
```
Comment (crowesn) [inline app/views/software_records/_form_server_environment.html.erb:28]: I want to make all of these "Yes|No" values boolean data types and switch to checkboxes. I just now noticed that our current pattern uses the controller to help track the values and I don't think it's sustainable.

File: app/controllers/software_records_controller.rb [hunk 2]
```diff
@@ -182,6 +182,14 @@ def software_record_params
       :cm_stakeholders,
       :cm_other_notes,
       :sensitive_information,
+      :qa_url,
+      :dev_url,
+      :production_server,
```
Comment (crowesn) [inline app/controllers/software_records_controller.rb:187]: Is this field for prod server url? Probably should match pattern with or url fields above

Comment (scherztc) [reply app/controllers/software_records_controller.rb:187]: Yeah and I think they need to be repeatable to represent a full stack application.

File: app/views/software_records/_form_server_environment.html.erb (hunk 1 above)
Comment (scherztc) [reply app/views/software_records/_form_server_environment.html.erb:28]: Use radio buttons with yes/no.   

Should some of that controller logic been in the model?

**Pull request merged**